03-transactions.md

```diff
@@ -158,22 +158,89 @@ Thus we use a simplified formula for *expected weight*, which assumes:
 
 The *expected weight* of a commitment transaction is calculated as follows:
 
-	transaction core: 4 + 1 + 1 + 4
-	transaction input: 32 + 4 + 1 + 4
-	transaction input witness: 1 + 74 + 74 + 1 + 1 + 34 + 34 + 1 + 1
-	transaction output: 8 + 1
-	transaction htlc output: 34
-	transaction to-local output: 34
-	transaction to-remote output: 25
-
+	p2wsh: 34 bytes
+		- OP_0: 1 byte
+		- OP_DATA: 1 byte (witness_script_SHA256 length)
+		- witness_script_SHA256: 32 bytes
+
+	p2wpkh: 22 bytes
+		- OP_0: 1 byte
+		- OP_DATA: 1 byte (public_key_HASH160 length)
+		- public_key_HASH160: 20 bytes
+
+	multi_sig: 71 bytes
+		- OP_2: 1 byte
+		- OP_DATA: 1 byte (pub_key_alice length)
+		- pub_key_alice: 33 bytes
+		- OP_DATA: 1 byte (pub_key_bob length)
+		- pub_key_bob: 33 bytes
+		- OP_2: 1 byte
+		- OP_CHECKMULTISIG: 1 byte
+
+	witness: 222 bytes
+		- number_of_witness_elements: 1 byte
+		- nil_length: 1 byte
+		- sig_alice_length: 1 byte
+		- sig_alice: 73 bytes
+		- sig_bob_length: 1 byte
+		- sig_bob: 73 bytes
+		- witness_script_length: 1 byte
+		- witness_script (multi_sig)
+		
+	funding_input: 41 bytes
+		- previous_out_point: 36 bytes
+			- hash: 32 bytes
+			- index: 4 bytes
+		- OP_DATA: 1 byte (ScriptSigLength)
+		- script_sig: 0 bytes
+		- witness <----	we use "Witness" instead of "ScriptSig" for
+	 			transaction validation, but "Witness" is stored
+	 			separately and cost for it size is smaller. So
+	 			we separate the calculation of ordinary data
+	 			from witness data.
+		- sequence: 4 bytes
+
+	output_paying_to_us: 43 bytes
+		- value: 8 bytes
+		- var_int: 1 byte (pk_script length)
+		- pk_script (p2wsh): 34 bytes
+
+	output_paying_to_them: 31 bytes
+		- value: 8 bytes
+		- var_int: 1 byte (pk_script length)
+		- pk_script (p2wpkh): 22 bytes
+
+	 htlc_output: 43 bytes
+		- value: 8 bytes
+		- var_int: 1 byte (pk_script length)
+		- pk_script (p2wsh): 34 bytes
+
+	 witness_header: 2 bytes
+		- flag: 1 byte
+		- marker: 1 byte
+
+	 commitment_transaction: 125 + 43 * num-htlc-outputs bytes
```

Comment:
I think this should actually be `125`: as `4+2+1+41+1+74+4 = 125` (expanding using the variables defined above).